engine/schema/src/main/java/com/cloud/dc/dao/VlanDaoImpl.java

```diff
@@ -167,7 +167,10 @@ public VlanDaoImpl() {
         List<PodVlanMapVO> vlanMaps = _podVlanMapDao.listPodVlanMapsByPod(podId);
         List<VlanVO> result = new ArrayList<VlanVO>();
         for (PodVlanMapVO pvmvo : vlanMaps) {
-            result.add(findById(pvmvo.getVlanDbId()));
+            VlanVO vlanByPodId = findById(pvmvo.getVlanDbId());
+            if (vlanByPodId != null && vlanByPodId.getRemoved() == null) {
```

Comment:
a null check is enough, removed items are not returned by `GenericDaoBase::findById()`

Reply:
I have added a commit addressing this comment. Thanks for the feedback, @rhtyd!